presto-main/src/test/java/com/facebook/presto/operator/scalar/TestArrayNormalizeFunction.java

```diff
@@ -0,0 +1,88 @@
+        assertFunction("array_normalize(ARRAY[-1.0E0, 2.0E0, 3.3E0], 0.0E0)", new ArrayType(DOUBLE), ImmutableList.of(-1.0, 2.0, 3.3));
+        assertFunction("array_normalize(ARRAY[REAL '-1.0', REAL '2.0', REAL '3.3'], REAL '0.0')", new ArrayType(REAL), ImmutableList.of(-1.0f, 2.0f, 3.3f));
+    }
+
+    @Test
+    public void test1Norm()
+    {
+        assertFunction("array_normalize(ARRAY[1.0E0, 2.0E0, 3.0E0], 1.0E0)", new ArrayType(DOUBLE), ImmutableList.of(1.0 / 6.0, 2.0 / 6.0, 3.0 / 6.0));
+        assertFunction("array_normalize(ARRAY[REAL '1.0', REAL '2.0', REAL '3.0'], REAL '1.0')", new ArrayType(REAL), ImmutableList.of(1.0f / 6.0f, 2.0f / 6.0f, 3.0f / 6.0f));
+
+        // Test with negative element.
+        assertFunction("array_normalize(ARRAY[-1.0E0, 2.0E0, 3.0E0], 1.0E0)", new ArrayType(DOUBLE), ImmutableList.of(-1.0 / 6.0, 2.0 / 6.0, 3.0 / 6.0));
+        assertFunction("array_normalize(ARRAY[REAL '-1.0', REAL '2.0', REAL '3.0'], REAL '1.0')", new ArrayType(REAL), ImmutableList.of(-1.0f / 6.0f, 2.0f / 6.0f, 3.0f / 6.0f));
+    }
+
+    @Test
+    public void test2Norm()
+    {
+        assertFunction("array_normalize(ARRAY[4.0E0, 3.0E0], 2.0E0)", new ArrayType(DOUBLE), ImmutableList.of(4.0 / 5.0, 3.0 / 5.0));
+        assertFunction("array_normalize(ARRAY[REAL '4.0', REAL '3.0'], REAL '2.0')", new ArrayType(REAL), ImmutableList.of(4.0f / 5.0f, 3.0f / 5.0f));
+
+        // Test with negative element.
+        assertFunction("array_normalize(ARRAY[-4.0E0, 3.0E0], 2.0E0)", new ArrayType(DOUBLE), ImmutableList.of(-4.0 / 5.0, 3.0 / 5.0));
+        assertFunction("array_normalize(ARRAY[REAL '-4.0', REAL '3.0'], REAL '2.0')", new ArrayType(REAL), ImmutableList.of(-4.0f / 5.0f, 3.0f / 5.0f));
+    }
+
+    @Test
+    public void testNulls()
+    {
+        assertFunction("array_normalize(null, 2.0E0)", new ArrayType(DOUBLE), null);
+        assertFunction("array_normalize(ARRAY[4.0E0, 3.0E0], null)", new ArrayType(DOUBLE), null);
+        assertFunction("array_normalize(ARRAY[4.0E0, null], 2.0E0)", new ArrayType(DOUBLE), null);
+        assertFunction("array_normalize(ARRAY[REAL '4.0', REAL '3.0'], null)", new ArrayType(REAL), null);
+    }
+
+    @Test
+    public void testUnsupportedType()
+    {
+        assertInvalidFunction("array_normalize(ARRAY[1, 2, 3], 1)", FUNCTION_IMPLEMENTATION_MISSING,
+                "Unsupported array element type for array_normalize function: integer");
+        assertInvalidFunction("array_normalize(ARRAY['a', 'b', 'c'], 'd')", FUNCTION_IMPLEMENTATION_MISSING,
+                "Unsupported type parameters.*");
+    }
+
+    @Test
+    public void testNegativeP()
+    {
+        assertInvalidFunction("array_normalize(ARRAY[1.0E0, 2.0E0, 3.3E0], -1.0E0)", INVALID_FUNCTION_ARGUMENT,
+                "array_normalize only supports non-negative p:.*");
+        assertInvalidFunction("array_normalize(ARRAY[REAL '1.0', REAL '2.0', REAL '3.3'], REAL '-1.0')", INVALID_FUNCTION_ARGUMENT,
```

Comment:
same; put the first param of a function to its own line if we need to break the function into multiple lines
```java
func(
    param1,
    param2,
    ...
    paramn);
```

Reply:
Done.